src/tools/archiveTools.cpp

```diff
@@ -80,37 +80,41 @@ bool getArchiveFavicon(const zim::Archive* const archive,
   return false;
 }
 
-std::string getMetaLanguage(const zim::Archive* const archive) {
+std::string getMetaLanguage(const zim::Archive& archive) {
   return getMetadata(archive, "Language");
 }
 
-std::string getMetaName(const zim::Archive* const archive) {
+std::string getMetaName(const zim::Archive& archive) {
   return getMetadata(archive, "Name");
 }
 
-std::string getMetaDate(const zim::Archive* const archive) {
+std::string getMetaDate(const zim::Archive& archive) {
   return getMetadata(archive, "Date");
 }
 
-std::string getMetaCreator(const zim::Archive* const archive) {
+std::string getMetaCreator(const zim::Archive& archive) {
   return getMetadata(archive, "Creator");
 }
 
-std::string getMetaPublisher(const zim::Archive* const archive) {
+std::string getMetaPublisher(const zim::Archive& archive) {
   return getMetadata(archive, "Publisher");
 }
 
-zim::Entry getFinalEntry(const zim::Archive* const archive, const zim::Entry& entry)
+zim::Item getFinalItem(const zim::Archive& archive, const zim::Entry& entry)
 {
-  return archive->getEntryByPath(entry.getItem(true).getPath());
+  return entry.getItem(true);
 }
 
-zim::Entry getEntryFromPath(const zim::Archive* const archive, const std::string& path)
+zim::Entry getEntryFromPath(const zim::Archive& archive, const std::string& path)
 {
-  if (path.empty() || path == "/") {
-    return archive->getMainEntry();
+  try {
+    return archive.getEntryByPath(path);
+  } catch (zim::EntryNotFound& e) {
+    if (path.empty()) {
```

Comment:
`path.empty() || path == "/"`

(I've told you that Reader behavior should not change, so use two different code path for reader and server.
But, I think we can share the new behavior, the change is pretty small and will not break existing usage of `Reader`)